### ISSUE TITLE
Perf: No point in checking java.lang.Object in RequestMappingRoutingTable.getAllRequestMappingsInHierarchy()

### ISSUE THREAD

**Issue by markkolich** [markkolich/curacao#15]
Performance profiling shows that countless milliseconds are wasted in Curacao's own `RequestMappingRoutingTable.getAllRequestMappingsInHierarchy()` method.  The logic therein involves a `while` loop which crawls the class hierarchy of a given `@Controller` looking for methods annotated with the `@RequestMapping` annotation:

```java
while (superClass != null) {
  // ...
}
```

This is fine, except that when we crawl up the class hierarchy and get to `java.lang.Object` we spend cycles looking through `java.lang.Object` for any methods annotated with `@RequestMapping`.  Good times, because there won't be any in `java.lang.Object` &mdash; why spend cycles using reflection checking `java.lang.Object` for any `@RequestMapping` annotated methods when there couldn't possibly be any?

We can clean this up by adding a simple check to the `while` loop in question:

```java
while (superClass != null && !Object.class.equals(superClass)) {
  // ...
}
```

Performance testing shows this shaves off at least 40ms per controller class (which may be many in large/complex web-applications).

**Comment by markkolich** [comment]
Upon further reflection (pun intended), we could probably simplify this to:

```java
while (!Object.class.equals(superClass)) {
  // ...
}
```

As we walk *up* the class hierarchy, we will eventually hit `java.lang.Object` and the `.equals()` check will catch `null` too.

**Comment by markkolich** [comment]
Fixed in version 2.8.6
https://github.com/markkolich/curacao/releases/tag/2.8.6
